rsyslog/go/src/github.com/mmexternal/undefined_field/undefined_field.go

```diff
@@ -0,0 +1,239 @@
+package main
+
+import (
+	"encoding/json"
+	"bufio"
+	"strconv"
+	"strings"
+	"fmt"
+	"os"
+)
+
+const (
+	initial_use_undefined              = false
+	initial_undefined_name             = "undefined"
+	initial_undefined_to_string        = false
+	initial_default_keep_fields        = "CEE,time,@timestamp,aushape,ci_job,collectd,docker,fedora-ci,file,foreman,geoip,hostname,ipaddr4,ipaddr6,kubernetes,level,message,namespace_name,namespace_uuid,offset,openstack,ovirt,pid,pipeline_metadata,rsyslog,service,systemd,tags,testcase,tlog,viaq_msg_id"
+	initial_extra_keep_fields          = ""
+	initial_keep_empty_fields          = ""
+	initial_undefined_max_num_fields   = -1
+	initial_undefined_dot_replace_char = "UNUSED"
+	initial_logging_file_path          = "/var/log/rsyslog/rsyslog.log"
+)
+
+var (
+	use_undefined bool
+	keep_fields []string
+	keep_empty_fields []string
+	undefined_name string
+	undefined_to_string bool
+	undefined_dot_replace_char string
+	undefined_max_num_fields int64
+	logfile *os.File
+	noaction          = false
+	replacer          = &strings.Replacer{}
+	basic_keep_fields = "_STREAM_ID,_SYSTEMD_INVOCATION_ID"
+)
+
+func onInit() {
+	// opening the rsyslog log file
+	logging_file_path := initial_logging_file_path
+	if eval := os.Getenv("LOGGING_FILE_PATH"); eval != "" {
+		logging_file_path = eval
+	}
+	if f, err := os.OpenFile(logging_file_path, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644); err == nil {
+		logfile = f
+	} else {
+		panic(err.Error())
+	}
+
+	// initializing
+	use_undefined = initial_use_undefined
+	default_keep_fields := initial_default_keep_fields
+	extra_keep_fields := initial_extra_keep_fields
+	undefined_name = initial_undefined_name
+	tmp_keep_empty_fields := initial_keep_empty_fields
+	undefined_to_string = initial_undefined_to_string
+	undefined_dot_replace_char = initial_undefined_dot_replace_char
+	undefined_max_num_fields = initial_undefined_max_num_fields
+
+	config, err := os.Open("/var/lib/rsyslog.pod/undefined.yaml")
```

Comment:
the go way to do this is to combine the assignment and the err check in a single statement
```go
if config, err := os.Open("/var/lib/rsyslog.pod/undefined.yaml"); err != nil {
    panic(fmt.Errorf("Could not open file [%s]: [%v]", "/var/lib/rsyslog.pod/undefined.yaml", err))
}
```

Reply:
Is it ok to allow the plugin to work without the undefined.yaml file with this warning in the log file?
`fmt.Fprintln(logfile, "WARNING: Could not open file [%s]: [%v]; Using the default configurations.", undefined_config, err)`
It might be a bit selfish, but the yaml file is generated by CLO pr https://github.com/openshift/cluster-logging-operator/pull/202, which depends upon this pr.  I'd like to make this pr having no dependency...

Reply:
so if the file is opened successfully, ok
if the file cannot be opened, because it does not exist, ok - no log is necessary
if the file cannot be opened, for some reason other than "does not exist", then panic with an error